test/distributed/org/apache/cassandra/distributed/test/accord/AccordHostReplacementTest.java

```diff
@@ -0,0 +1,104 @@
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import accord.utils.Gen;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
+import org.apache.cassandra.distributed.api.Feature;
+import org.apache.cassandra.distributed.api.TokenSupplier;
+import org.apache.cassandra.distributed.shared.ClusterUtils;
+import org.apache.cassandra.distributed.test.TestBaseImpl;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.service.consensus.TransactionalMode;
+import org.apache.cassandra.utils.CassandraGenerators;
+import org.apache.cassandra.utils.CassandraGenerators.TableMetadataBuilder;
+
+import static accord.utils.Property.qt;
+import static org.apache.cassandra.distributed.shared.ClusterUtils.stopUnchecked;
+import static org.apache.cassandra.utils.AccordGenerators.fromQT;
+import static org.apache.cassandra.utils.CassandraGenerators.insertCqlAllColumns;
+
+public class AccordHostReplacementTest extends TestBaseImpl
+{
+    private static final Logger logger = LoggerFactory.getLogger(AccordHostReplacementTest.class);
+
+    @Test
+    public void hostReplace() throws IOException
+    {
+        // start 2 node cluster and bootstrap 3rd node
+        // do a host replacement of one of the nodes
+        Cluster.Builder clusterBuilder = Cluster.build(3)
+                                                .withConfig(c -> c.with(Feature.values())
+                                                                  .set("accord.shard_count", "1"));
+        TokenSupplier tokenRing = TokenSupplier.evenlyDistributedTokens(3, clusterBuilder.getTokenCount());
+        int nodeToReplace = 2;
+        clusterBuilder = clusterBuilder.withTokenSupplier((TokenSupplier) node -> tokenRing.tokens(node == 4 ? nodeToReplace : node));
+        try (Cluster cluster = clusterBuilder.start())
+        {
+            fixDistributedSchemas(cluster);
+            init(cluster);
+
+            qt().withSeed(5215087322357346205L).withExamples(1).check(rs -> {
+                TableMetadata metadata = fromQT(new TableMetadataBuilder()
+                                                .withKeyspaceName(KEYSPACE)
+                                                .withTableKinds(TableMetadata.Kind.REGULAR)
+                                                .withKnownMemtables()
+                                                //TODO (coverage): include "fast_path = 'keyspace'" override
+                                                .withTransactionalMode(TransactionalMode.full)
+                                                .build())
+                                         .next(rs);
+                maybeCreateUDTs(cluster, metadata);
+                cluster.schemaChange(metadata.toCqlString(false, false));
+
+                Gen<ByteBuffer[]> dataGen = fromQT(fromQT(new CassandraGenerators.DataGeneratorBuilder(metadata).build(ignore -> 10)).next(rs));
```

Comment:
now that I have `ast.Statement` I could replace this and handle far more possible txn CQL

Reply:
spoke with Alex in slack, this might be a better place for Harry, as this class isn't testing the full domain of CQL, just host replacement and needs a load gen

Reply:
harry doesn't support accord timestamps... so harry here fails

```
Caused by: org.apache.cassandra.harry.model.Model$ValidationException: Timestamps in the row state don't match ones predicted by the model:
```